File: dynamic-programming/70-climbing-stairs/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:

        if n == 1 :
            return 1
        if n == 2 :
            return 2

        first = 1
        second = 2
        
        for x in range(3, n + 1):
            current = first + second
            first = second
            second = current
        
        return second

# Add this block to run the code locally in VS Code
if __name__ == "__main__":
    solver = Solution()
    # Replace with the test cases from the LeetCode problem
    result = solver.climbStairs(2)
    print(f"Output: {result}")

    result = solver.climbStairs(3)
    print(f"Output: {result}")

    result = solver.climbStairs(5)
    print(f"Output: {result}")

File: dynamic-programming/70-climbing-stairs/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {

    if (n === 1) return 1;
    if (n === 2) return 2;

    let first = 1;
    let second = 2;
    for (let x = 3; x <= n; x++) {
        let current = first + second;
        first = second;
        second = current;
    }

    return second;

};

console.log(climbStairs(2));
console.log(climbStairs(3));
console.log(climbStairs(5));